fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerTest.java

```diff
@@ -2620,4 +2636,25 @@ public void markReadable() {
             assertTrue(indexer.asyncToSync(FDBStoreTimer.Waits.WAIT_ONLINE_BUILD_INDEX, indexer.markReadableIfBuilt()));
         }
     }
+
+    @Test
+    public void testConfigLoader() {
+        Index index = runAsyncSetup();
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(index).setSubspace(subspace)
+                .setConfigLoader(old ->
+                        CompletableFuture.completedFuture(OnlineIndexer.Config.newBuilder()
+                                .setLimit(100)
+                                .setMaxRetries(3)
+                                .setRecordsPerSecond(10000)
+                                .build()))
+                .setMdcContext(ImmutableMap.of("mdcKey", "my cool mdc value"))
+                .setMaxAttempts(2)
+                .build()) {
+            indexBuilder.runAsync(vignore -> AsyncUtil.DONE, Pair::of, null, null);
+            assertEquals(indexBuilder.getLimit(), 100);
+            assertEquals(indexBuilder.getConfig().getMaxRetries(), 3);
+            assertEquals(indexBuilder.getConfig().getRecordsPerSecond(), 10000);
+        }
```

Comment:
I think this is missing a test that tries updating the config multiple times and validates that it does something reasonable each time (like that you get the correctly configured value each time).

It would also be good if there were a test that made sure that updating these values midstream were actually safe. I don't think there's one that I'm *particularly* worried about, though, so maybe it's fine as is.

Reply:
> I think this is missing a test that tries updating the config multiple times and validates that it does something reasonable each time (like that you get the correctly configured value each time).

I thought about this, but that required more visibility to some internals. I think it's a great idea. I added and exposed the number of times config is loaded and used it for testing. Take a look.


> It would also be good if there were a test that made sure that updating these values midstream were actually safe. I don't think there's one that I'm particularly worried about, though, so maybe it's fine as is.

I do not have enough context for this. I'll file a GitHub issue.